Sming/Wiring/IpAddress.h

```diff
@@ -132,36 +146,38 @@ class IpAddress : public Printable
 
 	bool compare(const IpAddress& addr, const IpAddress& mask) const
 	{
-		return ip_addr_netcmp(&address, &addr.address, &mask.address);
+		return ip4_addr_netcmp((ip4_addr_t* )&address, (ip4_addr_t* )&addr, (ip4_addr_t* )&mask);
 	}
 
 	// Overloaded index operator to allow getting and setting individual octets of the address
 	uint8_t operator[](int index) const
 	{
-		if(unsigned(index) >= sizeof(address)) {
+		if(unsigned(index) >= 4) {
```

Comment:
`sizeof(ip4_addr_t)` instead of magic number - see also other instances.

Reply:
> sizeof(ip4_addr_t) instead of magic number

Hm.. the size of IPv4 address is constant while the size of ip4_addr_t struct might not be...